### PR TITLE
Generated Alert Fields - Destinations Fix

### PR DESCRIPTION
**Pull request by wey-chiang**
## Background

Related to some issues discovered when debugging the FE implementation. 

The analysis-api test record expects a string to be returned from DDB for the destinationsOutput, however it will return a string set (since that's what the rules-engine will store in DDB)

## Changes

- Changed analysis-api model destinationsOutput field to be `[]string`
- Patched the `buildTestSubRecord` for `Destinations` with `buildTestSubRecordList` which handles the case of 

## Testing

- `mage test:ci`
- Deployment and manual testing in `us-east-1`

### CODE REVIEW

**Review by nhakmiller — APPROVED** [review]
LGTM, did you check with someone on the frontend that changing this response format is ok? And did you do a test deploy to a dev environment to verify?

**Comment by nhakmiller** [comment]
You also need the `sync-to-release` label.

**Comment by wey-chiang** [comment]
> LGTM, did you check with someone on the frontend that changing this response format is ok? And did you do a test deploy to a dev environment to verify?

From @vorillaz:
> The graphql is expecting that the output is always a string

The changes take the list and concatenates all of the strings with a comma separator -- so the end result is compatible with exactly what is expected.

dest-override-works!

File: internal/core/analysis_api/analysis/rule_engine.go [hunk 1]
```diff
@@ -135,6 +137,22 @@ func buildTestSubRecord(output, error string) *models.TestDetectionSubRecord {
...
 
+func buildTestSubRecordList(output []string, error string) *models.TestDetectionSubRecord {
+	if len(output) == 0 && error == "" {
+		return nil
+	}
+
+	result := &models.TestDetectionSubRecord{}
+
+	if len(output) != 0 {
+		result.Output = aws.String(strings.Join(output, ","))
```
Comment (wey-chiang) [inline internal/core/analysis_api/analysis/rule_engine.go:148]:
```suggestion
		result.Output = aws.String(strings.Join(output, ", "))
```

**Review by austinbyers — APPROVED** [review]

**Pull request merged**